core/GlobalState.cc

```diff
@@ -1194,13 +1188,14 @@ void moveNames(pair<unsigned int, unsigned int> *from, pair<unsigned int, unsign
     }
 }
 
-void GlobalState::expandNames(int growBy) {
+void GlobalState::expandNames(u4 newSize) {
```

Comment:
Do you have any idea why `expandNames` requires all this extra stuff, but expanding the symbol table is just a matter of a quick call to `reserve`?

It looks like this function does nothing more than re-implement `vector::reserve`...

Reply:
This is the source of `vector::reserve` in `libc++`:

```cpp
template <class _Tp, class _Allocator>
void
vector<_Tp, _Allocator>::reserve(size_type __n)
{
    if (__n > capacity())
    {
        allocator_type& __a = this->__alloc();
        __split_buffer<value_type, allocator_type&> __v(__n, size(), __a);
        __swap_out_circular_buffer(__v);
    }
}
```

Reply:
My vote would be to replace all calls `expandNames` with just calls to `names.reserve(names.capacity() * 2)`.

Reply:
`namesByHash` is a vector used like a map where the size is used to determine the key, so it's more than a simple `reserve` unfortunately (look at `lookupNameUnique`). The resizing operation re-keys every entry in the old vector to a new key (index) in the new vector.

Reply:
gotcha thanks! i did not see that `moveNames` was doing so much work (:shakes_fist: collapsed portions of the diff)